ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/FourActivity.java

```diff
@@ -0,0 +1,33 @@
+package net.boondockradio.activitiesandintents;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.widget.TextView;
+
+public class FourActivity extends AppCompatActivity {
+
+    public static void startActivity(Context context) {
+        Intent intent = new Intent(context, OneActivity.class);
+        intent.putExtra("message", "on moved 4");
```

Comment:
one thing we normally do here is use a `static final String`

``` java
private static final String KEY_MESSAGE = "message";

// startActivity
intent.putExtra(KEY_MESSAGE, "on moved 4");

// onCreate
String message = intent.getStringExtra(KEY_MESSAGE);
```

For me, main reason is no typo ie no mistakes.
Also autocomplete if we use `KEY_MESSAGE`